environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentLoadBalancerService.java

```diff
@@ -63,4 +63,14 @@ public void updateLoadBalancerInEnvironmentAndStacks(EnvironmentDto environmentD
                     environmentLbDto.getEndpointAccessGateway(), environmentLbDto.getEndpointGatewaySubnetIds(), userCrn);
         }
     }
+
+    private boolean isLoadBalancerEnabledForDatalake(String accountId, String cloudPlatform) {
+        return entitlementService.datalakeLoadBalancerEnabled(accountId) ||
+                isLoadBalancerEntitlementNotRequiredForCloudProvider(cloudPlatform) ||
+                entitlementService.publicEndpointAccessGatewayEnabled(accountId);
```

Comment:
nit: It would be more efficient to have the `isLoadBalancerEntitlementNotRequiredForCloudProvider(cloudPlatform)` line be the first check, because it's a simple string compare and the cheapest operation.